addons/web_editor/static/src/scss/web_editor.frontend.scss

```diff
@@ -55,6 +55,10 @@
 
 .container-fluid > .o_grid_mode {
     --gutter-x: 30px;
+
+    .o_grid_item > .row {
+        margin-right: calc(-1 * var(--grid-item-padding-x, 0));
+    }
```

Comment:
Put this just before the `.o_grid_item_image {` rule. It should not be in the scope of `.container-fluid > .o_grid_mode`, the bug is the same for other container than `.container-fluid`, even if there is no overflow/scrollbar in those cases.